python/run-tests.py

```diff
@@ -96,7 +96,8 @@ def run_individual_python_test(target_dir, test_name, pyspark_python):
     os.mkdir(metastore_dir)
 
     # Also override the JVM's temp directory by setting driver and executor options.
-    java_options = "-Djava.io.tmpdir={0} -Dio.netty.tryReflectionSetAccessible=true".format(tmp_dir)
+    java_options = "-Djava.io.tmpdir={0}".format(tmp_dir)
+    java_options = java_options + " -Dio.netty.tryReflectionSetAccessible=true -Xss16M"
```

Comment:
The -Xss16m seems OK, just wondering if we think that breeze 2.0 is going to cause stack overflows more readily without this setting? could be an issue

Reply:
yes, without this setting, pyspark and sparkr always failed.

I think it should be a new restriction in breeze, found such a commit in breeze 2.0
https://github.com/scalanlp/breeze/commit/07c2d06dcb05928ac5c665d4312236cbcbd8a0aa

Reply:
but I guess configurations in practice should normally safisty this stack size requirement.
for example, the scala UT run with breeze 2.0 successfully.

Reply:
also cc @HyukjinKwon @WeichenXu123 since this PR change the java option of tests in pySpark and sparkR.

Reply:
Are we using smaller memory in `PySpark` and `SparkR`?

Scala UT seems to use small stack size, `4m`, at large memory like `mx4g`.
```
-Xmx4g -Xss4m -XX:MaxMetaspaceSize=2g
```

Although this is already passed, do you think we can use more consistent settings across Scala/Python/R?

Reply:
let me have a try with `-Xss4m`